dali/benchmark/operator_bench.h

```diff
@@ -120,17 +120,21 @@ class OperatorBench : public DALIBenchmark {
     CUDA_CALL(cudaStreamSynchronize(0));
 
     int64_t batches = 0;
-    for (auto _ : st) {
+
+    while (true) {
+      if (!st.KeepRunning()) {
+        // If no iterations left, synchronize and exit the loop
+        CUDA_CALL(cudaStreamSynchronize(0));
+        break;
+      }
+
       op_ptr->Run(ws);
       batches++;
+
       if (sync_each_n > 0 && batches % sync_each_n == 0) {
         CUDA_CALL(cudaStreamSynchronize(0));
       }
     }
-
-    st.ResumeTiming();
```

Comment:
Note: This triggers an assert in debug build. Apparently ResumeTiming/PauseTiming are meant to be used within the iteration loop